document/value.go

```diff
@@ -483,11 +483,12 @@ func calculateFloats(a, b Value, operator byte) (res Value, err error) {
 
 		return NewDoubleValue(xa / xb), nil
 	case '%':
-		if xb == 0 {
+		ia, ib := int64(xa), int64(xb)
+
+		if ib == 0 {
 			return NewNullValue(), nil
 		}
 
-		ia, ib := int64(xa), int64(xb)
 		return NewDoubleValue(float64(ia % ib)), nil
```

Comment:
We should use `math.Mod` to compute the remainder of xa/xb.

Special cases (from Go doc):
```
Mod(±Inf, y) = NaN
Mod(NaN, y) = NaN
Mod(x, 0) = NaN
Mod(x, ±Inf) = x
Mod(x, NaN) = NaN
```

Reply:
Special cases are already handled by `math.Mod`, so I think we should just map NaN to NullValue on return.

Reply:
You are absolutely right, I focused in solving described issue. WDYT about adding this improvements in a new PR?

Reply:
I noticed this PR make no sense since new implementation will overwrite it. I'll do this changes in current PR.

Reply:
I think it’s fine to overwrite changes in this PR.

Reply:
@tie done